src/__tests__/unit/mcp-controller.unit.ts

```diff
@@ -0,0 +1,278 @@
+import {
+  expect,
+  sinon,
+  StubbedInstanceWithSinonAccessor,
+} from '@loopback/testlab';
+
+import {Request, Response} from '@loopback/rest';
+
+import {McpServerFactory} from '../../services';
+
+import {ILogger, STATUS_CODE} from '@sourceloop/core';
+
+import {Server} from '@modelcontextprotocol/sdk/server/index.js';
+
+import {StreamableHTTPServerTransport} from '@modelcontextprotocol/sdk/server/streamableHttp.js';
+
+import {McpController} from '../../controllers';
+
+describe('McpController (unit)', () => {
+  let controller: McpController;
+
+  let logger: StubbedInstanceWithSinonAccessor<ILogger>;
+
+  let serverFactory: StubbedInstanceWithSinonAccessor<McpServerFactory>;
+
+  let mockServer: sinon.SinonStubbedInstance<Server>;
+
+  let mockTransport: sinon.SinonStubbedInstance<StreamableHTTPServerTransport>;
+
+  let req: Request;
+
+  let res: sinon.SinonStubbedInstance<Response>;
+  let handlers: {[key: string]: Function[]};
+
+  beforeEach(() => {
+    // Mock logger
+
+    logger = {
+      info: sinon.stub(),
+      error: sinon.stub(),
+      warn: sinon.stub(),
+      debug: sinon.stub(),
+    } as unknown as StubbedInstanceWithSinonAccessor<ILogger>;
+
+    // Mock server
+
+    mockServer = {
+      connect: sinon.stub().resolves(),
+      close: sinon.stub().resolves(),
+    } as unknown as sinon.SinonStubbedInstance<Server>;
+
+    // Mock transport
+    mockTransport = {
+      handleRequest: sinon.stub().resolves(),
+      close: sinon.stub().resolves(),
+    } as unknown as sinon.SinonStubbedInstance<StreamableHTTPServerTransport>;
+
+    // Mock server factory
+    serverFactory = {
+      createServer: sinon.stub().returns(mockServer),
+    } as unknown as StubbedInstanceWithSinonAccessor<McpServerFactory>;
+
+    // Mock request
+    req = {
+      body: {jsonrpc: '2.0', method: 'test'},
+    } as Request;
+
+    // Mock response
+    res = {
+      // Initialize the array to hold captured handlers
+      on: sinon.stub().callsFake((event: string, handler: Function) => {
+        if (!handlers[event]) {
+          handlers[event] = [];
+        }
+        handlers[event].push(handler);
+        return res;
+      }),
+      headersSent: false,
+      status: sinon.stub().returnsThis(),
+      json: sinon.stub().returnsThis(),
+    } as unknown as sinon.SinonStubbedInstance<Response>;
+
+    handlers = {};
+    // Stub StreamableHTTPServerTransport constructor
+
+    sinon
+      .stub(StreamableHTTPServerTransport.prototype, 'handleRequest')
+      .callsFake(mockTransport.handleRequest);
+
+    sinon
+      .stub(StreamableHTTPServerTransport.prototype, 'close')
+      .callsFake(mockTransport.close);
+
+    controller = new McpController(logger, serverFactory);
+  });
+
+  afterEach(() => {
+    sinon.restore();
+  });
+
+  describe('handleMCPRequest', () => {
+    it('should successfully handle MCP request', async () => {
```

Comment:
this should be an acceptance test with actual mcp client to ensure real coverage